temporal/api/enums/v1/activity_state.proto

```
@@ -0,0 +1,54 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package temporal.api.enums.v1;
+
+option go_package = "go.temporal.io/api/enums/v1;enums";
+option java_package = "io.temporal.api.enums.v1";
+option java_multiple_files = true;
+option java_outer_classname = "ActivityStateProto";
+option ruby_package = "Temporalio::Api::Enums::V1";
+option csharp_namespace = "Temporalio.Api.Enums.V1";
+
+// More detailed state of an activity.
+enum ActivityState {
+    ACTIVITY_STATE_UNSPECIFIED = 0;
+    // Activity was scheduled but not yet started.
+    ACTIVITY_STATE_SCHEDULED = 1;
+    // Activity has started executing.
+    ACTIVITY_STATE_STARTED = 2;
+    // Activity was requested to be canceled.
+    ACTIVITY_STATE_CANCEL_REQUESTED  = 3;
```

I am a little concerned about this as a state. I think this is better as a boolean on the `ActivityInfo`. While I understand it's a state and not an event (in the sense that you can't send a second cancel), I think there's two benefits to a bool on the info:

1. Because this is really `ACTIVITY_STATE_RUNNING_AND_CANCEL_REQUESTED` which is a bit confusing I can't just check for "running" to see if running
2. There is value in even knowing a cancel was requested regardless of whether swallowed (unsure we have this information for completed activities though)

I am curious about a behavior though. What if I have an activity that is requested to cancel but ignores/swallows it returns a general activity failure? Does it retry? Or is it marked failed? I will ask internally because it affects my opinion here.

---

I add it because it was there in pending activity state. I agree - I will remove the state and add a flag to activity info. It will be true only for pending activities that received cancel request.